cmd/opentelemetry-collector/app/exporter/cassandra/factory.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cassandra
+
+import (
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector/config/configerror"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/plugin/storage/cassandra"
+)
+
+const (
+	// TypeStr defines type of the Cassandra exporter.
+	TypeStr = "jaeger_cassandra"
+)
+
+// OptionsFactory returns initialized cassandra.OptionsFactory structure.
+type OptionsFactory func() *cassandra.Options
+
+// DefaultOptions creates Cassandra options supported by this exporter.
+func DefaultOptions() *cassandra.Options {
+	return cassandra.NewOptions("cassandra")
+}
+
+// Factory is the factory for Jaeger Cassandra exporter.
+type Factory struct {
+	OptionsFactory OptionsFactory
```

Comment:
in what scenarios does this need to be a function instead of a struct?

Reply:
We need to initialize the `Options` struct with wiper once the collector service has started.

We can revisit this and instead pass a viper instance and run the initialization in `CreateDefaultConfgi`.